### PR TITLE
rsx: zcull fix, cellGcm regression fix after #7262, FIFO recovery improvements

### PR DESCRIPTION
**Pull request by elad335**
* Fix size argument passed to `rsx::thread::read_barrier()` in image_in.
* Fix a typo in cellGcm HLE which led to memory size and io address to be x100000 as smaller as they should.
* Make fifo call/ret recovery from unexpected stack state be the same as invalid cmd recovery, add logging for its cmd hex representation in this case.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellGcmSys.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ u32 gcmIoOffsetToAddress(u32 ioOffset)
 {
 	const u32 upper12Bits = g_fxo->get<gcm_config>()->offsetTable.eaAddress[ioOffset >> 20];
 
-	if (static_cast<s16>(upper12Bits) < 0)
+	if (upper12Bits <= 0xBFF)
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellGcmSys.cpp]: Actually there is a bug here, don't merge yet.

**Comment by elad335** [comment]
Resolved.

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 2]
```diff
@@ -828,15 +828,30 @@ namespace rsx
 	{
 		// Get timestamp, and convert it from microseconds to nanoseconds
 		const u64 t = get_guest_system_time() * 1000;
-		if (t != timestamp_ctrl)
+
+		if (t <= vtimestamp)
```
Comment (kd-11) [inline rpcs3/Emu/RSX/RSXThread.cpp]: This will murder zcull when we have a long queue (some games can queue almost a thousand entries at once). I suggest moving this to another PR for now as we think of better solutions.

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 3]
```diff
@@ -1000,7 +1000,7 @@ namespace rsx
 			else
 			{
 				const u32 data_length = in_pitch * (in_h - 1) + src_line_length;
-				rsx->read_barrier(src_address, dst_address, true);
+				rsx->read_barrier(src_address, data_length, true);
```
Comment (kd-11) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: Nice catch! That's a facepalm-worthy error from me XD
